ev3dev2/stopwatch.py

```diff
@@ -9,76 +9,113 @@
 else:
     import datetime as dt
 
-
 def get_ticks_ms():
     if is_micropython():
         return utime.ticks_ms()
     else:
         return int(dt.datetime.timestamp(dt.datetime.now()) * 1000)
 
-
 class StopWatch(object):
-
+    """
+    A timer class which lets you start timing and then check the amount of time
+    elapsed.
+    """
     def __init__(self, desc=None):
+        """
+        Initializes the StopWatch but does not start it.
+
+        desc:
+            A string description to print when stringifying.
+        """
         self.desc = desc
-        self._value = 0
-        self.start_time = None
-        self.prev_update_time = None
+        self._start_time = None
+        self._stopped_total_time = None
 
     def __str__(self):
-        if self.desc is not None:
-            return self.desc
-        else:
-            return self.__class__.__name__
+        name = self.desc if self.desc is not None else self.__class__.__name__
+        return "{}: {}".format(name, self.hms_str)
 
     def start(self):
-        assert self.start_time is None, "%s is already running" % self
-        self.start_time = get_ticks_ms()
-
-    def update(self):
-
-        if self.start_time is None:
-            return
-
-        current_time = get_ticks_ms()
-
-        if self.prev_update_time is None:
-            delta = current_time - self.start_time
-        else:
-            delta = current_time - self.prev_update_time
-
-        self._value += delta
-        self.prev_update_time = current_time
+        """
+        Starts the timer. If the timer is already running, resets it.
+        """
+        self._stopped_total_time = None
```

Comment:
If the timer is already running and they call start() I think we should raise an exception. That feels like a error scenario we should make the user aware of.